TF_Service_dotNet/TouchFree_Service/Connection/Services/WebSocketReceiver.cs

```diff
@@ -190,6 +218,232 @@ void HandleGetStatusRequest(string _content)
             clientMgr.SendStatus(currentConfig);
         }
 
+        #region DiagnosticAPI_Requests
+        void HandleTrackingRequest(IncomingRequest _request)
+        {
+            JObject contentObj = JsonConvert.DeserializeObject<JObject>(_request.content);
+
+            // Explicitly check for requestID else we can't respond
+            if (!RequestIdExists(contentObj))
+            {
+                string message = "Tracking State change request failed. This is due to a missing or invalid requestID";
+
+                var maskResponse = new SuccessWrapper<MaskingData?>(false, message, null);
+                var boolResponse = new SuccessWrapper<bool?>(false, message, null);
+
+                TrackingApiState state = new TrackingApiState() {
+                    requestID =  "",
+                    mask = maskResponse,
+                    allowImages = boolResponse,
+                    cameraReversed = boolResponse,
+                    analyticsEnabled= boolResponse
+                };
+
+                // This is a failed request, do not continue with processing the request,
+                // the Client will have no way to handle the config state
+                clientMgr.SendTrackingState(state);
+                return;
+            }
+
+            _request.requestId = contentObj.GetValue("requestID").ToString();
+
+            if (_request.action == ActionCode.GET_TRACKING_STATE)
+            {
+                HandleGetTrackingStateRequest(_request);
+            }
+            else
+            {
+                HandleSetTrackingStateRequest(contentObj, _request);
+            }
+        }
+
+        void HandleGetTrackingStateRequest(IncomingRequest _request)
+        {
+            trackingApiResponse = new TrackingResponse(_request.requestId, _request.content, true, true, true, true, true);
+            responseOriginTime = DateTime.Now.Millisecond;
```

Comment:
This just gets the milliseconds of the time not the time as milliseconds. I think we need something else to set the origin time